src/Orchard.Environment.Shell.Data/ShellDescriptorManager.cs

```diff
@@ -31,14 +32,20 @@ public class ShellDescriptorManager : IShellDescriptorManager
             _eventBus = eventBus;
             _session = session;
             _logger = logger;
+
+            _asyncLazyShellDescriptor = new Lazy<Task<ShellDescriptor>>(() => Task.Run(() =>
```

Comment:
Now when running the tool shape tag helper are resolved then the following is called
GetShapeTable() => GetEnabledFeaturesAsync() => GetCurrentShell() => GetShellDescriptorAsync().

But the tool use a `Parallel.Foreach()` to compile views, so the above is used in parallel. No problem with `ShapeTableManager` which is transient, but there is a database access issue in `ShellDescriptorManager.GetShellDescriptorAsync()`. So, i needed this change to fix it.

Another solution is to be able to use in this context the `AllFeaturesShellDescriptorManager` (with no db query) but didn't find a way to replace this service through the scope provider which is not a services collection. See at the end of my 1st comment in this PR.